spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelAutoConfiguration.java

```diff
@@ -76,49 +80,63 @@
 
 	@Bean
 	@ConditionalOnMissingBean
-	OpenTelemetry otel(TracerProvider tracerProvider, ContextPropagators contextPropagators) {
-		OpenTelemetry openTelemetry = DefaultOpenTelemetry.builder().setTracerProvider(tracerProvider)
+	OpenTelemetry otel(SdkTracerProvider tracerProvider, ContextPropagators contextPropagators) {
+		// this is super hacky. it would be better to go and eliminate all usage of the
+		// global. That requires some changes upstream in the instrumentation APIs, so for
+		// now, we hack.
+		GlobalOpenTelemetry.resetForTest();
+		OpenTelemetrySdk openTelemetrySdk = OpenTelemetrySdk.builder().setTracerProvider(tracerProvider)
 				.setPropagators(contextPropagators).build();
-		GlobalOpenTelemetry.set(openTelemetry);
-		return openTelemetry;
+		GlobalOpenTelemetry.set(openTelemetrySdk);
+		return openTelemetrySdk;
 	}
 
 	@Bean
 	@ConditionalOnMissingBean
-	TracerProviderFactory otelTracerProviderFactory() {
-		return new SdkTracerProviderFactory();
+	SdkTracerProvider otelTracerProvider(TraceConfig traceConfig, ObjectProvider<List<SpanProcessor>> spanProcessors,
+			SpanExporterCustomizer spanExporterCustomizer, ObjectProvider<List<SpanExporter>> spanExporters,
+			Sampler sampler, Resource resource) {
```

Comment:
What is this `Resource` ?

Reply:
Not sure I understand the question. Can you clarify?